packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -909,11 +902,9 @@ abstract class TableBase extends Resource implements ITable {
   ): iam.Grant {
     if (opts.tableActions) {
       const resources = [this.tableArn,
-        Lazy.string({ produce: () => this.hasIndex ? `${this.tableArn}/index/*` : Aws.NO_VALUE }),
```

Comment:
Is the `hasIndex` field used anywhere else in this class? Can we just remove it now?

Reply:
and drop the `Lazy`s (everything is known at construction time now).

Reply:
I removed hasIndex and the Lazy's.